File: ios/StockPro/Data/DTOs/WarehouseDTOs.swift
import Foundation

struct WarehouseDTO: Decodable, Identifiable {
    let id: Int
    let name: String
    let address: String?
    let manager: String?
    let phone: String?
    let is_default: Int?
    let created_at: String?

    func toItem() -> WarehouseListItem {
        WarehouseListItem(
            id: id,
            name: name,
            address: address ?? "",
            manager: manager ?? "",
            phone: phone ?? "",
            isDefault: (is_default ?? 0) == 1
        )
    }
}

struct WarehouseCreateRequest: Encodable {
    let name: String
    let address: String?
    let manager: String?
}

struct WarehouseListItem: Identifiable {
    let id: Int
    let name: String
    let address: String
    let manager: String
    let phone: String
    let isDefault: Bool
}

File: App2/core/data/src/main/java/com/app2/core/data/remote/dto/WarehouseDTOs.kt
package com.app2.core.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class WarehouseDTO(
    val id: Int,
    val name: String,
    val address: String? = null,
    val manager: String? = null,
    val phone: String? = null,
    @SerialName("is_default") val isDefault: Int? = null,
    val ice: String? = null,
    val patente: String? = null,
    val rc: String? = null,
    @SerialName("taxe_number") val taxeNumber: String? = null,
    @SerialName("created_at") val createdAt: String? = null
)

@Serializable
data class WarehouseCreateRequest(
    val name: String,
    val address: String? = null,
    val manager: String? = null,
    val phone: String? = null,
    val ice: String? = null,
    val patente: String? = null,
    val rc: String? = null,
    @SerialName("taxe_number") val taxeNumber: String? = null
)
